src/lib/WIFI/devWIFI.cpp

```diff
@@ -281,59 +262,101 @@ static void HandleReset(AsyncWebServerRequest *request)
+      json["config"]["button-actions"][button][pos]["action"] = actions[pos].action;
+    }
   }
+#endif
+
+  json["config"]["ssid"] = station_ssid;
+  json["config"]["mode"] = wifiMode == WIFI_STA ? "STA" : "AP";
   #if defined(TARGET_RX)
-  s += ",\"modelid\":" + String(config.GetModelId());
-  s += ",\"forcetlm\":" + String(config.GetForceTlmOff());
-  #endif
+  json["config"]["modelid"] = config.GetModelId();
+  json["config"]["forcetlm"] = config.GetForceTlmOff();
   #if defined(GPIO_PIN_PWM_OUTPUTS)
-  if (GPIO_PIN_PWM_OUTPUTS_COUNT > 0) {
-    s += WebGetPwmStr();
+  for (uint8_t ch=0; ch<GPIO_PIN_PWM_OUTPUTS_COUNT; ++ch)
+  {
+    json["config"]["pwm"][ch] = config.GetPwmChannel(ch)->raw;
   }
   #endif
-  s += ",\"product_name\": \"" + String(product_name) + "\"";
-  s += ",\"lua_name\": \"" + String(device_name) + "\"";
-  s += ",\"reg_domain\": \"" + String(getRegulatoryDomain()) + "\"";
-  s += ",\"uid\": [";
-  for (int i=0 ; i<6 ; i++) {
-    if (i!=0) s += ",";
-    s += String(UID[i], 10);
-  }
-  s += + "]";
-  s += ",\"uidtype\": ";
+  #endif
+  json["config"]["product_name"] = product_name;
+  json["config"]["lua_name"] = device_name;
+  json["config"]["reg_domain"] = getRegulatoryDomain();
+  JsonArray uid = json["config"].createNestedArray("uid");
+  copyArray(firmwareOptions.uid, sizeof(firmwareOptions.uid), uid);
+
   #if defined(TARGET_RX)
-  if (config.GetOnLoan()) s += "\"On loan\"";
+  if (config.GetOnLoan()) json["config"]["uidtype"] = "On loan";
   else
   #endif
-  if (firmwareOptions.hasUID) s += "\"Flashed\"" ;
+  if (firmwareOptions.hasUID) json["config"]["uidtype"] = "Flashed";
   #if defined(TARGET_RX)
-  else if (config.GetIsBound()) s += "\"Traditional\"";
-  else s += "\"Not set\"";
+  else if (config.GetIsBound()) json["config"]["uidtype"] = "Traditional";
+  else json["config"]["uidtype"] = "Not set";
   #else
-  else s += "\"Not set (using MAC address)\"";
+  else json["config"]["uidtype"] = "Not set (using MAC address)";
   #endif
-  s += "}";
-  request->send(200, "application/json", s);
+
+  AsyncResponseStream *response = request->beginResponseStream("application/json");
+  serializeJson(json, *response);
+  request->send(response);
+}
+
+#if defined(TARGET_TX)
+static void UpdateConfiguration(AsyncWebServerRequest *request, JsonVariant &json)
+{
+  if (json.containsKey("button-actions")) {
+    const JsonArray &array = json["button-actions"].as<JsonArray>();
+    for (int button=0 ; button<array.size() ; button++)
+    {
+      tx_button_color_t action;
+      for (int pos=0 ; pos<MAX_BUTTON_ACTIONS ; pos++)
+      {
+        action.val.actions[pos].pressType = array[button][pos]["is-long-press"];
+        action.val.actions[pos].count = array[button][pos]["count"];
+        action.val.actions[pos].action = array[button][pos]["action"];
+      }
+      config.SetButtonActions(button, &action);
+    }
+    config.Commit();
+  }
+  request->send(204);
```

Comment:
This sends 204 back according to devtools but there's no cuteAlert generated to say it was successful.

Reply:
I'll fix it in the .js file.